api/client.go

```diff
@@ -579,21 +588,19 @@ func (c *Client) NewRequest(method, requestPath string) *Request {
 		lookupPath = requestPath
 	}
 
-	req.MFAHeaderVals = c.mfaCreds
+	req.MFAHeaderVals = mfaCreds
 
-	if c.wrappingLookupFunc != nil {
-		req.WrapTTL = c.wrappingLookupFunc(method, lookupPath)
+	if wrappingLookupFunc != nil {
+		req.WrapTTL = wrappingLookupFunc(method, lookupPath)
 	} else {
 		req.WrapTTL = DefaultWrappingLookupFunc(method, lookupPath)
 	}
-	if c.config.Timeout != 0 {
-		c.config.HttpClient.Timeout = c.config.Timeout
```

Comment:
Is this intentionally omitted?

Reply:
It is! Because this was racy, and one of the reasons to change this in the first place. The client is reused so every request was re-setting this timeout value (potentially), and by the time that it even got to making the request it could have been modified by another user. Instead we read the value while locked, give up the lock, then use a context with a timeout during the actual request making.

Reply:
:+1: